sdk/dotnet/Pulumi/Testing/Mocks.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Pulumi.Testing
+{
+    public interface IMocks
+    {
+        Task<object> CallAsync(string token, object args, string? provider);
+
+        Task<(string id, object state)> NewResourceAsync(string type, string name, object inputs, string? provider,
+            string? id);
+    }
+    
+    class MockMonitor : IMonitor
+    {
+        private readonly IMocks mocks;
+        public readonly List<string> LoggedMessages = new List<string>();
+        
+        public MockMonitor(IMocks mocks)
+        {
+            this.mocks = mocks;
+        }
+
+        public Task<global::Pulumirpc.InvokeResponse> InvokeAsync(global::Pulumirpc.InvokeRequest request)
+            => throw new NotImplementedException();
+
+        public Task<global::Pulumirpc.ReadResourceResponse> ReadResourceAsync(
+            global::Pulumirpc.ReadResourceRequest request)
+            => throw new NotImplementedException();
+
+        public Task<global::Pulumirpc.RegisterResourceResponse> RegisterResourceAsync(
+            global::Pulumirpc.RegisterResourceRequest request)
+            => throw new NotImplementedException();
```

Comment:
@CyrusNajmabadi I would appreciate your help in implementing the above three methods to pass calls `IMocks`

Reply:
I took a stab on that, resolving this question